spigot/src/main/java/io/github/retrooper/packetevents/util/SpigotConversionUtil.java

```diff
@@ -109,4 +118,49 @@ public static ItemStack fromBukkitItemStack(org.bukkit.inventory.ItemStack itemS
     public static org.bukkit.inventory.ItemStack toBukkitItemStack(ItemStack itemStack) {
         return SpigotReflectionUtil.encodeBukkitItemStack(itemStack);
     }
+
+    public static NBTCompound fromNmsNbt(Object nbtCompound) {
+        byte[] bytes;
+        try (ByteArrayOutputStream byteStream = new ByteArrayOutputStream();
+             DataOutputStream stream = new DataOutputStream(byteStream)) {
+            SpigotReflectionUtil.writeNmsNbtToStream(nbtCompound, stream);
+            bytes = byteStream.toByteArray();
+        } catch (IOException e) {
+            e.printStackTrace();
+            return null;
+        }
+
+        Object buffer = UnpooledByteBufAllocationHelper.wrappedBuffer(bytes);
+        PacketWrapper<?> wrapper = PacketWrapper.createUniversalPacketWrapper(buffer);
+        NBTCompound nbt = wrapper.readNBT();
+        ByteBufHelper.release(buffer);
+        return nbt;
+    }
+
+    public static Object toNmsNbt(NBTCompound nbtCompound) {
+        Object buffer = UnpooledByteBufAllocationHelper.buffer();
+        PacketWrapper<?> wrapper = PacketWrapper.createUniversalPacketWrapper(buffer);
+        wrapper.writeNBT(nbtCompound);
+        byte[] bytes = ByteBufHelper.copyBytes(buffer);
+        ByteBufHelper.release(buffer);
+        try (ByteArrayInputStream byteStream = new ByteArrayInputStream(bytes);
+             DataInputStream stream = new DataInputStream(byteStream)) {
+            return SpigotReflectionUtil.readNmsNbtFromStream(stream);
+        } catch (IOException e) {
+            e.printStackTrace();
+            return null;
+        }
+    }
+
+    public static Dimension fromBukkitWorld(World world) {
+        boolean v1_19 = PacketEvents.getAPI().getServerManager().getVersion().isNewerThanOrEquals(ServerVersion.V_1_19);
+        boolean v1_16_2 = PacketEvents.getAPI().getServerManager().getVersion().isNewerThanOrEquals(ServerVersion.V_1_16_2);
+        if (v1_19 || !v1_16_2) {
+            return new Dimension(DimensionType.getById(world.getEnvironment().getId()));
+        } else {
+            Object worldServer = SpigotReflectionUtil.convertBukkitWorldToWorldServer(world);
+            Object nbt = SpigotReflectionUtil.convertWorldServerDimensionToNmsNbt(worldServer);
```

Comment:
This will execute for 1.16.2 servers but the Methods used in`convertWorldServerDimensionToNmsNbt` will only be initialised on 1.17+ servers.

Reply:
It only seems that only versions 1.17 and 1.18 need nbt serialization

Reply:
According to [this](https://github.com/retrooper/packetevents/blob/b090424510df32686e6ab2fc04dd46d00f334e4c/api/src/main/java/com/github/retrooper/packetevents/wrapper/PacketWrapper.java#L707), the method would work perfectly

Reply:
I have tried using this on 1.16.2 and it calls `convertWorldServerDimesntionToNmsNbt` which will not work because the methods required are only initialised if the server 1.17+ as you can see [here](https://github.com/retrooper/packetevents/pull/451/files/8eb7eae82646d21d5282f759b76a26acfcf5e90c#diff-4d7cfd5213e9b6871aade33ad1e127b5fa933d60e5eea545a27979960d463f2aR174-R181).

Reply:
Fixed for all version ig